atlassian_jwt_auth/aio/verifier.py

```diff
@@ -0,0 +1,24 @@
+import asyncio
+
+from atlassian_jwt_auth.verifier import JWTAuthVerifier as _JWTAuthVerifier
+
+
+class JWTAuthVerifier(_JWTAuthVerifier):
+    def __init__(self, public_key_retriever, *, loop=None, **kwargs):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+        self.loop = loop
+        super().__init__(public_key_retriever, **kwargs)
+
+    async def _decode_jwt(self, a_jwt, key_id, jwt_key='', **kwargs):
+        """Decode JWT and check if it's valid
+
+        Args:
+            a_jwt (bytes): serialized JWT token.
+            key_id (str): `kid` parameter of the token.
+            jwt_key (str | asyncio.Future): JWT public key or a future that
+                provides the key.
+        """
+        if asyncio.iscoroutine(jwt_key):
```

Comment:
It was the easiest way to await for `atlassian_jwt_auth.aio.HTTPSPublicKeyRetriever.retrieve` while reusing the synchronous code of `atlassian_jwt_auth.verifier`.

Do you have a suggestion how it can be done in a better way?